File: forest/endangered-burl.lua
loop
    local enemy = self:findNearestEnemy()
    if enemy.type == "burl" then
        self:say("I'm not attacking that Burl!")
    elseif enemy.type == "munchkin" or enemy.type == "thrower" then
        self:attack(enemy)
    elseif enemy.type == "ogre" then
        self:moveXY(21, 42)
    end
end

File: forest/endangered-burl.clj
(while true
  (def e (.findNearestEnemy this))
  (if e
    (if (= (.type e) "burl")
      (.say this "I'm not attacking that Burl!")
      (if (= (.type e) "ogre")
        (.moveXY this 21 39)
        (.attack this e)))
    (.wait this 0.1)))
